pkg/scalers/scalersconfig/typed_config.go

```diff
@@ -478,6 +492,25 @@ func (sc *ScalerConfig) configParamValue(params Params) (string, bool) {
 	return "", params.IsNested()
 }
 
+// checkUnexpectedParameterExist is a function that checks if there are any unexpected parameters in the TriggerMetadata
+func (sc *ScalerConfig) checkUnexpectedParameterExist(parsedParamNames []string, logger logr.Logger) {
+	for k := range sc.TriggerMetadata {
+		suffix := "FromEnv"
+		if !strings.HasSuffix(k, "FromEnv") {
+			suffix = ""
+		}
+		key := strings.TrimSuffix(k, suffix)
+		if !slices.Contains(parsedParamNames, key) {
+			if sc.Recorder != nil {
+				message := fmt.Sprintf("Unmatched input property %s in scaler %s", key+suffix, sc.ScalableObjectType)
+				// Just logging as it's optional property checking and should not block the scaling
+				logger.Error(nil, message)
+				sc.Recorder.Event(sc.ScaledObject, corev1.EventTypeNormal, eventreason.KEDAScalersInfo, message)
```

Comment:
Should we use `corev1.EventTypeWarning`  instead of `corev1.EventTypeNormal` here?

Reply:
I could see a good rationale for either type. My usual rule of thumb is
* use `Normal` when something successfully happened and a user should know about it - e.g. volume is mounted, some connection was established...
* use `Warning` when something unexpected happened and it has negative impact for the user - e.g. volume didn't get mounted, some connection got broken...

This case "an extra parameter user provided is ignored". KEDA is successful at ignoring it but it might have unexpected result for the user, so a bit of both. I'd probably also pick `Warning` event but I can be easily convinced for `Normal`.

Reply:
In my mind, this is a warning more than a normal event. We are adding logic to detect an unexpected configuration, so IMHO is something to warn about to the user (as we have considered that checking it is important enough to do it)